ADAL/src/ADAuthenticationContext.m

```diff
@@ -234,7 +236,17 @@ + (BOOL)isResponseFromBroker:(NSString *)sourceApplication
     isBroker = isBroker || [sourceApplication isEqualToString:ADAL_BROKER_APP_BUNDLE_ID_DOGFOOD];
 #endif
     
-    return response && isBroker;
+    NSURLComponents *components = [NSURLComponents componentsWithURL:response resolvingAgainstBaseURL:NO];
+    NSString *qp = [components percentEncodedQuery];
+    NSDictionary *queryParamsMap = [NSDictionary adURLFormDecode:qp];
+    
+    NSString *protocolVersion = queryParamsMap[@"msg_protocol_ver"];
+    BOOL isValidVersion = [protocolVersion isEqualToString:@"2"];
+    
+    NSDictionary *resumeDictionary = [[NSUserDefaults standardUserDefaults] objectForKey:kAdalResumeDictionaryKey];
+    BOOL isADALInitiatedRequest = [resumeDictionary[kAdalSDKNameKey] isEqualToString:kAdalSDKObjc];
+    
+    return response && isBroker && isValidVersion && isADALInitiatedRequest;
```

Comment:
Nit: method name "isResponseFromBroker" indicates we're checking if it's a response from broker only. Now we're also checking if response is for us. Can we rename the method or add a separate one indicating its purpose?

Reply:
Added a separate method.